File: string-handling/substrings/python/in_string.py
# Raspberry Pi Foundation
# Developed to be used alongside Isaac Computer Science,
# part of the National Centre for Computing Education
# Usage licensed under CC BY-SA 4

def check_in_string():
    """Demonstrates checking if a substring occurs within a string"""
    my_string = "Hello World!"
    is_in_string = "World" in my_string
    print(is_in_string)
    

# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == '__main__':
    check_in_string()

File: string-handling/substrings/c-sharp/in_string.cs
/*
Raspberry Pi Foundation
Developed to be used alongside Isaac Computer Science, part of the National Centre for Computing Education
Usage licensed under CC BY-SA 4

Note: This file is designed to be copied out and compiled on your machine.
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file.
To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Visual Studio, for example)
3. Change the namespace to match your project (if necessary)
4. Compile the program
5. Run the program
*/

using System;

namespace IsaacCodeSamples
{

    class Substrings
    {

        // The Main method is the entry point for all C# programs
        public static void Main() {
            CheckInString();
        }


        // Demonstrates checking if a substring occurs within a string
        public static void CheckInString() {
            string myString = "Hello World!";
            bool isInString = myString.Contains("World");
            Console.WriteLine(isInString);
        }


    }
}
